### PR TITLE
feat: fetch all NO from contracts

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: src/web3_extentions/lido_validators.py [hunk 1]
```diff
@@ -64,16 +58,21 @@ def get_lido_validators_by_node_operators(self, blockstamp: BlockStamp) -> Dict[
...
-                )
-                for operator in module.operators
-            ])
+                    stakingModuleAddress=module.module.stakingModuleAddress,
+                    validators=[
+                        v for v in lido_validators
+                        if v.key.moduleAddress == module.module.stakingModuleAddress
+                        and v.key.operatorIndex == operator.index
+                    ],
+                ),  # it's tuple, and it will be faster than append or extend or concatenation with list
```
Comment (skhomuti) [inline src/web3_extentions/lido_validators.py:75]: looks like over-engineering, imho just make append

File: src/providers/keys/typings.py [hunk 2]
```diff
@@ -51,6 +58,7 @@ class Operator:
 @dataclass
 class OperatorExpanded(Operator):
     stakingModuleAddress: Address
+    validators: list[LidoValidator]
```
Comment (F4ever) [inline src/providers/keys/typings.py:61]: Remove validators from OperatorExpanded

File: src/web3_extentions/lido_validators.py [hunk 3]
```diff
@@ -63,17 +74,22 @@ def get_lido_validators_by_node_operators(self, blockstamp: BlockStamp) -> Dict[
...
 
-        for module in operators_by_modules:
-            operators.extend([
-                OperatorExpanded(
-                    stakingModuleAddress=module.module.stakingModuleAddress,
-                    **asdict(operator),
-                )
-                for operator in module.operators
-            ])
+        staking_modules = self.w3.lido_contracts.staking_router.functions.getStakingModules().call(
```
Comment (F4ever) [inline src/web3_extentions/lido_validators.py:80]: Let's move contract call to separate method and add there comment with response type

File: src/web3_extentions/lido_validators.py [hunk 4]
```diff
@@ -1,25 +1,36 @@
...
+    targetValidatorsCount: int
+    stuckValidatorsCount: int
+    refundedValidatorsCount: int
+    stuckPenaltyEndTimestamp: int
+    validatorsReport: tuple[int, int, int]  # totalExited, totalDeposited, depositable
 
 
-NodeOperatorIndex = Tuple[Address, int]
+@dataclass
+class OperatorExpanded(Operator):
```
Comment (F4ever) [inline src/web3_extentions/lido_validators.py:32]: Let's remove this class. It was created to support key's api response

File: src/web3_extentions/lido_validators.py [hunk 5]
```diff
@@ -1,25 +1,60 @@
...
     from src.web3_extentions.typings import Web3
 
+NodeOperatorIndex = Tuple[Address, int]
+
 
 @dataclass
-class LidoValidator:
-    key: LidoKey
-    validator: Validator
+class StakingModule(Nested):
```
Comment (F4ever) [inline src/web3_extentions/lido_validators.py:21]: here is no need in Nested

File: src/web3_extentions/lido_validators.py [hunk 6]
```diff
@@ -1,25 +1,60 @@
...
+    lastDepositAt: int
+    # block.number of the last deposit of the staking module
+    lastDepositBlock: int
+    # number of exited validators
+    exitedValidatorsCount: int
 
 
-NodeOperatorIndex = Tuple[Address, int]
+@dataclass
+class NodeOperator:
```
Comment (F4ever) [inline src/web3_extentions/lido_validators.py:46]: Add Nested here

**Comment by F4ever** [comment]
Lets remove LidoValidator from Keys.typings to web3_extentions.lido_validators

**Comment by F4ever** [comment]
Merged into consensus-class

**Pull request closed without merging**